ranch/ranch.go

```diff
@@ -174,21 +176,7 @@ func (r *Ranch) Acquire(rType, state, dest, owner, requestID string) (*crds.Reso
 		}
 
 		if new {
-			logger.Debug("Checking for associated dynamic resource type...")
-			lifeCycle, err := r.Storage.GetDynamicResourceLifeCycle(rType)
-			// Assuming error means no associated dynamic resource.
-			if err == nil {
-				if typeCount < lifeCycle.Spec.MaxCount {
-					logger.Debug("Adding new dynamic resources...")
-					res := newResourceFromNewDynamicResourceLifeCycle(r.Storage.generateName(), lifeCycle, r.now())
-					if err := r.Storage.AddResource(res); err != nil {
-						logger.WithError(err).Warningf("unable to add a new resource of type %s", rType)
-					}
-					logger.Infof("Added dynamic resource %s of type %s", res.Name, res.Spec.Type)
-				}
-			} else {
-				logrus.WithError(err).Debug("Failed listing DRLC")
-			}
+			addResource(logger, r, rType, typeCount)
```

Comment:
why was this factored out into its own function?

Reply:
To avoid https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_boskos/40/pull-boskos-build-test-verify/1305588541102755842#1:build-log.txt%3A223